dlib/cmake_utils/find_blas.cmake

```diff
@@ -67,7 +76,11 @@ if (UNIX OR MINGW)
    pkg_check_modules(BLAS_REFERENCE cblas)
    pkg_check_modules(LAPACK_REFERENCE lapack)
    if (BLAS_REFERENCE_FOUND AND LAPACK_REFERENCE_FOUND)
-      set(blas_libraries "${BLAS_REFERENCE_LDFLAGS}")
+      if (NOT HAVE_CBLAS)
+         set(blas_libraries cblas blas)
+      else()
+         set(blas_libraries "${BLAS_REFERENCE_LDFLAGS}")
+      endif()
```

Comment:
This isn't right.  It should be something like this

```
SET(CMAKE_REQUIRED_LIBRARIES "${BLAS_REFERENCE_LDFLAGS}")
CHECK_FUNCTION_EXISTS(cblas_ddot HAVE_CBLAS)
if (BLAS_REFERENCE_FOUND AND LAPACK_REFERENCE_FOUND AND HAVE_CBLAS)
   set(blas_libraries "${BLAS_REFERENCE_LDFLAGS}")
```

That is, just check if what `pkgconfig` found is right.  If it isn't then don't go into the if statement.  What you added is now saying "if pkgconfig found something that isn't right, just blindly assume cblas and blas libraries exist and work".  But they may well not work.

If pkgconfg failed to find blas this if should fall through and we should proceed onto the rest of the test code in this file and let it locate a good blas library.  If that doesn't work for you then you need to change something further down in the file.

Reply:
Thanks for advisory Davis, Today I have learned, will try fix again.